src/init.cpp

```diff
@@ -820,10 +811,13 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     LogPrintf("Using at most %i connections (%i file descriptors available)\n", nMaxConnections, nFD);
     std::ostringstream strErrors;
 
-    LogPrintf("Using %u threads for script verification\n", nScriptCheckThreads);
-    // BU: parallel block validation - begin
-    AddAllScriptCheckQueuesAndThreads(nScriptCheckThreads, &threadGroup); // This initializes and creates 4 separate script thread queues and thread pools.
-    // BU: parallel block validation - end
+    // -par=0 means autodetect, but passing 0 to the CParallelValidation constructor means no concurrency
+    int nPVThreads = GetArg("-par", DEFAULT_SCRIPTCHECK_THREADS);
+    if (nPVThreads <= 0)
+        nPVThreads += GetNumCores();
```

Comment:
I would not change this section...from what it was...i remember looking at this as well once and the issue is from what i remember, more subtle than at first glance and requires more thought.  I think it would be good to address this one in a separate PR.  (the original is pretty goofy and non intuitive though i must say)